streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -529,40 +529,53 @@ private long computeLatency() {
 
         try {
             records = consumer.poll(pollTimeMs);
-        } catch (NoOffsetForPartitionException ex) {
-            TopicPartition partition = ex.partition();
-            if (builder.earliestResetTopicsPattern().matcher(partition.topic()).matches()) {
-                log.info(String.format("stream-thread [%s] setting topic to consume from earliest offset %s", this.getName(), partition.topic()));
-                consumer.seekToBeginning(ex.partitions());
-            } else if (builder.latestResetTopicsPattern().matcher(partition.topic()).matches()) {
-                consumer.seekToEnd(ex.partitions());
-                log.info(String.format("stream-thread [%s] setting topic to consume from latest offset %s", this.getName(), partition.topic()));
-            } else {
-
-                if (originalReset == null || (!originalReset.equals("earliest") && !originalReset.equals("latest"))) {
-                    setState(State.PENDING_SHUTDOWN);
-                    String errorMessage = "No valid committed offset found for input topic %s (partition %s) and no valid reset policy configured." +
-                        " You need to set configuration parameter \"auto.offset.reset\" or specify a topic specific reset " +
-                        "policy via KStreamBuilder#stream(StreamsConfig.AutoOffsetReset offsetReset, ...) or KStreamBuilder#table(StreamsConfig.AutoOffsetReset offsetReset, ...)";
-                    throw new StreamsException(String.format(errorMessage, partition.topic(), partition.partition()), ex);
-                }
+        } catch (final InvalidOffsetException e) {
+            final Set<TopicPartition> partitions = e.partitions();
```

Comment:
I'm thinking this block should be extracted to a method. It doesn't really have much to do with `pollRequests` and it would be nicer to read if there was a method like `resetPositions` or similar